src/V3EmitCImp.cpp

```diff
@@ -705,33 +768,43 @@ class EmitCTrace final : EmitCFunc {
             if (AstEnumDType* const enump = VN_CAST(nodep->skipRefToEnump(), EnumDType)) {
                 int enumNum = m_enumNumMap[enump];
                 if (!enumNum) {
+                    if (typesSplitNeeded()) {
```

Comment:
Please move all of this into a new function that returns the enum number, making it if need be.